generators/app/templates/test/_karma.conf.js

```diff
@@ -11,7 +19,7 @@ module.exports = function(config) {
 
   // If no browsers are specified, we enable `karma-detect-browsers`
   // this will detect all browsers that are available for testing
-  if (!config.browsers.length) {
+  if (config.browsers !== false && !config.browsers.length) {
```

Comment:
passing `--no-browsers` causes this to be false rather than an empty array, and we want to respect `--no-browsers` for the `test:server`